File: 45_jump_game_II.cxx
classSolution {
public:
    int jump(vector<int>& nums) {
        if(nums.size()==1) return 0;
        int cnt=0;
        for(int jump=0,loc=0,i=0; i<nums.size()-1 && loc<nums.size()-1; ++i) {
            jump = max(i+nums[i],jump);
            if(i==loc) {
                loc=jump;
                cnt++;
            } 
        }
        return cnt;
    }
};

File: 45_jump_game_II.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        if len(nums)==1: return 0
        cnt, jump, loc = 0, 0, 0
        for i,v in enumerate(nums[:-1]):
            jump = max(i+v,jump)
            if i==loc:
                cnt+=1
                if loc==(len(nums)-1): break
                loc=jump
        return cnt
